frontend/public/components/modals/create-namespace-modal.jsx

```diff
@@ -129,20 +129,63 @@ const CreateNamespaceModalWithTranslation = connect(
         return (
           <>
             <p>{nameFormat}</p>
-            {this.props.createProject ? <p>{createNamespace}</p> : null}
+            {createProject ? <p>{createNamespace}</p> : null}
           </>
         );
       };
+
+      const popoverDescriptionText = (
+        <>
+          <p>
+            <strong>{t('public~OpenShift Projects and Kubernetes Namespaces')}</strong>
+          </p>
+          <p>
+            {t(
+              'public~Projects are exposed as editable to end users while namespaces are not. Direct creation of a project is typically restricted to administrators, while end users should use the requestproject resource.',
+            )}
+          </p>
+          <p>
+            <a
+              className="co-external-link"
+              href="https://docs.openshift.com/online/pro/architecture/core_concepts/projects_and_users.html"
+              rel="noopener noreferrer"
+              target="_blank"
+            >
+              {t('public~Learn more about OpenShift Projects and Namespaces')}
+            </a>
+          </p>
+        </>
+      );
       return (
         <form
           onSubmit={this._submit.bind(this)}
           name="form"
           className="modal-content modal-content--no-inner-scroll"
         >
           <ModalTitle>
-            {this.props.createProject ? t('public~Create Project') : t('public~Create Namespace')}
+            {createProject ? t('public~Create Project') : t('public~Create Namespace')}
           </ModalTitle>
           <ModalBody>
+            <p>
+              {createProject
+                ? t(
+                    'public~An OpenShift project is an alternative representation of a Kubernetes namespace.',
+                  )
+                : t(
+                    'public~A Kubernetes namespace is an alternative representation of an OpenShift project.',
```

Comment:
We should only show this message if it's on an OpenShift cluster. On a native Kubernetes cluster, we shouldn't mention projects. We can check this using the `OPENSHIFT` feature flag.